### PR TITLE
Adding new TOS to Rewards UI Opt-In points

### PR DESCRIPTION
**Pull request by ryanml**
Related: https://github.com/brave/brave-browser/issues/3124

This adds block of TOS copy to the points Rewards can be opted in to specified in the issue. This also adds a new component called `DisabledBox`.

With the Pre-Opt-In rewards panel, some padding from the design was reduced to respect vertical height.

Welcome Page: 
Screen Shot 2019-04-01 at 5 49 56 PM

Pre-Opt-In Panel:
Screen Shot 2019-04-01 at 5 49 32 PM

Rewards Setting Page (With new component)
Screen Shot 2019-04-01 at 5 49 15 PM

Welcome Page Header:
Screen Shot 2019-04-05 at 2 59 21 AM

##### Link / storybook path to visual changes
https://brave-ui-klmmryl7y.now.sh

## Integration
- [ ] Does this contain changes to src/components or src/
  - [ ] Will you publish to npm immediately after this PR, or wait until sometime in the future?
  - [ ] Incompatible API change to something existing _(major version increase)_
  - [ ] Adding new backwards-compatible functionality? _(minor version increase)_
  - [ ] Fixing a bug backwards-compatibly? _(patch version increase)_
  
- [ ] Does this contain changes to src/features for brave-core?
  - [ ] Are there non backwards-compatible changes required for brave-core? **Do not merge until brave-core PR is approvable.** Link to brave-core PR:
  - [ ] Will you create brave-core PR to update to this commit after it is merged?
  - [ ] Wants uplift to brave-core feature branch?
     - When uplift-approved, merge to brave-core-0.VV.x feature branch
     - Create additional brave-core PRs for each feature branch to update commit

### CODE REVIEW

File: src/features/rewards/disabledBox/index.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+export default class DisabledBox extends React.PureComponent<{}, {}> {
+  render () {
+    return (
+      <StyledWrapper>
+        <StyledCard>
+          <StyledTitle>
+            {getLocale('whyBraveRewards')}
+          </StyledTitle>
+          <StyledText padding={true}>
```
Comment (NejcZdovc) [inline src/features/rewards/disabledBox/index.tsx]: why is text padding from title different?

File: src/features/rewards/disabledBox/style.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+export const StyledWrapper = styled<{}, 'div'>('div')`
+  display: block;
+  width: 100%;
+  margin-bottom: 28px;
+`
+
+export const StyledCard = styled(Card)`
+  font-family: Poppins, sans-serif;
+  border-bottom-left-radius: 6px;
```
Comment (NejcZdovc) [inline src/features/rewards/disabledBox/style.ts]: why is this needed as `Card` already has border radius se 6px for all sides?

File: src/features/rewards/disabledBox/style.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  color: #442DC8;
+  font-size: 22px;
+  font-weight: 600;
+  letter-spacing: 0;
+  line-height: 28px;
+  padding-bottom: 5px;
+`
+
+export const StyledText = styled<StyleProps, 'p'>('p')`
+  color: #6B6B79;
```
Comment (NejcZdovc) [inline src/features/rewards/disabledBox/style.ts]: let's use theme color

File: src/features/rewards/disabledBox/style.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+export const StyledCard = styled(Card)`
+  font-family: Poppins, sans-serif;
+  border-bottom-left-radius: 6px;
+  border-bottom-right-radius: 6px;
+`
+
+export const StyledTitle = styled<{}, 'span'>('span')`
+  width: 541px;
+  color: #442DC8;
```
Comment (NejcZdovc) [inline src/features/rewards/disabledBox/style.ts]: let's use theme color

File: src/features/rewards/disabledBox/style.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+export const StyledWrapper = styled<{}, 'div'>('div')`
+  display: block;
+  width: 100%;
+  margin-bottom: 28px;
+`
+
+export const StyledCard = styled(Card)`
+  font-family: Poppins, sans-serif;
```
Comment (NejcZdovc) [inline src/features/rewards/disabledBox/style.ts]: let's use theme font definition

File: src/features/rewards/disabledBox/style.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+`
+
+export const StyledText = styled<StyleProps, 'p'>('p')`
+  color: #6B6B79;
+  font-size: 16px;
+  font-weight: normal;
+  letter-spacing: 0;
+  line-height: 26px;
+  padding-bottom: ${p => p.padding ? 10 : 0}px;
+  font-family: Muli, sans-serif;
```
Comment (NejcZdovc) [inline src/features/rewards/disabledBox/style.ts]: let's use theme font definition

File: src/features/rewards/mainToggle/style.ts [hunk 7]
```diff
@@ -47,26 +48,27 @@ export const StyledTM = styled<{}, 'span'>('span')`
...
-  font-size: 16px;
-  font-weight: 300;
-  line-height: 1.75;
-  color: #838391;
 `
 
-export const StyledContent = styled<{}, 'div'>('div')`
-  flex-basis: 100%;
+export const StyledServiceText = styled<{}, 'span'>('span')`
+  color: #4A4A4A;
```
Comment (NejcZdovc) [inline src/features/rewards/mainToggle/style.ts]: let's use theme color

File: src/features/rewards/panelWelcome/index.tsx [hunk 8]
```diff
@@ -79,42 +91,51 @@ export default class PanelWelcome extends React.PureComponent<Props, {}> {
...
-              this.props.creating && !this.props.error
-              ? <StyledJoinButton
-                level='secondary'
+          {
+            this.props.creating && !this.props.error
+            ? <StyledJoinButton
+              level='secondary'
+              size='call-to-action'
+              type='subtle'
+              text={'Creating wallet'}
```
Comment (NejcZdovc) [inline src/features/rewards/panelWelcome/index.tsx]: please add localisation for this one

File: src/features/rewards/panelWelcome/style.ts [hunk 9]
```diff
@@ -100,3 +101,24 @@ export const StyledErrorMessage = styled<{}, 'span'>('span')`
...
+export const StyledTOSWrapper = styled<{}, 'div'>('div')`
+  display: block;
+  margin: 20px auto;
+  width: 300px;
+`
+
+export const StyledServiceText = styled<{}, 'span'>('span')`
+  color: rgba(255, 255, 255, 0.65);
+  font-size: 14px;
+  font-family: Muli, sans-serif;
```
Comment (NejcZdovc) [inline src/features/rewards/panelWelcome/style.ts]: please use theme font definition

File: src/features/rewards/welcomePage/style.ts [hunk 10]
```diff
@@ -196,3 +196,24 @@ export const StyledAlertLeft = styled<{}, 'div'>('div')`
...
+export const StyledTOSWrapper = styled<{}, 'div'>('div')`
+  width: 325px;
+  display: block;
+  margin: 20px auto 0;
+`
+
+export const StyledServiceText = styled<{}, 'span'>('span')`
+  color: rgba(0, 0, 0, 0.65);
+  font-size: 12px;
+  font-family: Muli, sans-serif;
```
Comment (NejcZdovc) [inline src/features/rewards/welcomePage/style.ts]: please use theme font definition

File: src/features/rewards/welcomePage/style.ts [hunk 11]
```diff
@@ -196,3 +196,24 @@ export const StyledAlertLeft = styled<{}, 'div'>('div')`
...
+  font-weight: normal;
+  letter-spacing: 0;
+  line-height: 18px;
+`
+
+export const StyledServiceLink = styled<{}, 'a'>('a')`
+  cursor: pointer;
+  color: #000000;
```
Comment (NejcZdovc) [inline src/features/rewards/welcomePage/style.ts]: please use theme color

File: src/features/rewards/panelWelcome/style.ts (hunk 9 above)
Comment (ryanml) [reply src/features/rewards/panelWelcome/style.ts]: @NejcZdovc we use this exact rule (`font-family: Muli, sans-serif;`) across may different components, what is the definition you are expecting to see?

File: src/features/rewards/disabledBox/style.ts (hunk 2 above)
Comment (ryanml) [reply src/features/rewards/disabledBox/style.ts]: Rules removed

File: src/features/rewards/disabledBox/index.tsx (hunk 1 above)
Comment (ryanml) [reply src/features/rewards/disabledBox/index.tsx]: The desired effect is now done with CSS, no props

File: src/features/rewards/disabledBox/style.ts (hunk 3 above)
Comment (ryanml) [reply src/features/rewards/disabledBox/style.ts]: done

File: src/features/rewards/disabledBox/style.ts (hunk 4 above)
Comment (ryanml) [reply src/features/rewards/disabledBox/style.ts]: done

File: src/features/rewards/mainToggle/style.ts (hunk 7 above)
Comment (ryanml) [reply src/features/rewards/mainToggle/style.ts]: done

File: src/features/rewards/panelWelcome/index.tsx (hunk 8 above)
Comment (ryanml) [reply src/features/rewards/panelWelcome/index.tsx]: done, we already have an entry for this one (`braveRewardsCreatingText`), it just wasn't being used.

File: src/features/rewards/welcomePage/style.ts (hunk 11 above)
Comment (ryanml) [reply src/features/rewards/welcomePage/style.ts]: done

File: src/features/rewards/mainToggle/index.tsx [hunk 12]
```diff
@@ -22,11 +22,20 @@ export interface Props {
   onToggle: () => void
   id?: string
   testId?: string
+  onTOSClick: () => void
```
Comment (NejcZdovc) [inline src/features/rewards/mainToggle/index.tsx]: in other places you set this two as optional, can you please set it here as well?

Comment (ryanml) [reply src/features/rewards/mainToggle/index.tsx]: done
